### PR TITLE
Car sharing examples

### PR DESCRIPTION
**Pull request by huksley**
Added car sharing request and response examples and into the Open API specification

- [x] Example payloads for request and response
- [x] Tests for schema validation for payloads
- [x] Embedded into the documentation in description and endpoints
- [x] Links to geojson and image will work after merging

demo here: https://huksley.github.io/maas-tsp-api/redoc.html

### CODE REVIEW

File: examples/car/booking-create-request.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    },
+    "from": {
+      "lat": 60.1693,
+      "lon": 24.9364,
+      "name": "EuroPark P-Simonkenttä",
+      "address": "streetName:Simonkatu|streetNumber:5|zipCode:00100|city:Helsinki"
+    },
+    "mode": "SHARED_CAR",
+    "endTime": 1576607463612,
+    "agencyId": "CARSHARINGABC",
```
Comment (MichalCz) [inline examples/car/booking-create-request.json]: AgencyId names were in usually in lower-dash-case...

File: examples/car/booking-update-request-lock.json [hunk 2]
```diff
@@ -1,4 +1,4 @@
 {
-  "tspId": "deadbeef-deadbeef-deadbeef-beefdead-121111-211211",
+  "tspId": "64c09cf9-cd7d-422e-9d42-8ab897bab9d5",
```
Comment (MichalCz) [inline examples/car/booking-update-request-lock.json]: Is this random, or is that an actual tsp id on some stage?

Comment (huksley) [reply examples/car/booking-update-request-lock.json]: This is random but the same through examples, generated by uuidgen on command line

File: examples/car/booking-create-request.json (hunk 1 above)
Comment (huksley) [reply examples/car/booking-create-request.json]: Fixed

**Pull request merged**